browser/src/react-redux/App/SignupPage.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+
+const initialState = {
+  confirmPassword: '',
+  email: '',
+  password: '',
+  emailError: '',
+  generalError: '',
+  passwordError: ''
+};
+
+class signupinput extends Component {
+  state = initialState
+
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  getConfirmPasswordError = () => {
+    return (this.state.confirmPassword && this.state.password !== this.state.confirmPassword)
+      ? 'Passwords do not match.'
+      : null;
+  }
+
+  enableButton = () => {
+    return (
+      this.state.confirmPassword &&
+      this.state.password &&
+      this.state.confirmPassword &&
+      this.state.password === this.state.confirmPassword
+    )
+      ? false
```

Comment:
You don't need a ternary here. You can just return the conditional checks above. Remember, that will already give you a true or false.